modules/internal/String.chpl

```diff
@@ -867,6 +878,27 @@ module String {
       return localThis.buff[i - 1];
     }
 
+    /*
+      :returns: The value of a single-codepoint string as an integer.
+     */
+    proc toCodepoint(): int(32) {
+      var localThis: string = this.localize();
+
+      if localThis.isEmpty() then
+        halt("string.toCodepoint() only accepts single-codepoint strings");
+
+      var cp: int(32);
+      var nbytes: c_int;
+      var multibytes = localThis.buff: c_string;
+      var maxbytes = localThis.len: ssize_t;
+      qio_decode_char_buf(cp, nbytes, multibytes, maxbytes);
```

Comment:
OOC, why isn't the return value of `qio_decode_char_buf()` checked here or in the rest of the file?  It looks like it has error cases that can leave `nbytes` looking reasonable.

Reply:
In expected operation with UTF8 strings (which is all this is supposed to handle), it will return an error code *and* set the codepoint to the unicode replacement character. That's reasonable in some ways. But beyond that, the design is that a `string` can only hold valid UTF-8, so if there is an invalid UTF-8 character in it, that should have been raised as an error somewhere in the process of creating the string.

Reply:
I thought about checking the return code, but I ended up not doing it because of the reasons Michael gave, and I convinced myself that `nbytes` would return what I wanted even in an error condition.